scipy/spatial/transform/rotation.pyx

```diff
@@ -1357,6 +1459,82 @@ cdef class Rotation(object):
 
         return angles[0] if self._single else angles
 
+    def as_mrp(self):
+        """Represent as Modified Rodrigues Parameters (MRPs).
+
+        MRPs is a 3 dimensional vector co-directional to the axis of rotation and whose
+        magnitude is equal to tan(theta / 4), where theta is the angle of rotation (in radians) [1].
+
+        MRPs have a singuarity at 360 degrees which can be avoided by ensuring the angle of
+        rotation does not exceed 180 degrees, i.e. switching the direction of the rotation when
+        it is past 180 degrees. This function will always return MRPs corresponding to a rotation
+        of less than or equal to 180 degrees.
+
+        Returns
+        -------
+        mrps : ndarray, shape (3,) or (N, 3)
+            Shape depends on shape of inputs used for initialization.
+
+        References
+        ----------
+        .. [1] Shuster, M. D. "A Survery of Attitude Representations",
+               The Journal of Astronautical Sciences, Vol. 41, No.4, 1993,
+               pp. 475-476
+
+        Examples
+        --------
+        >>> from scipy.spatial.transform import Rotation as R
+
+        Represent a single rotation:
+
+        >>> r = R.from_rotvec([0, 0, np.pi])
+        >>> r.as_mrp()
+        array([0.        , 0.        , 1.         ])
+        >>> r.as_mrp().shape
+        (3,)
+
+        Represent a stack with a single rotation:
+
+        >>> r = R.from_euler('xyz', [[180, 0, 0]], degrees=True)
+        >>> r.as_mrp()
+        array([[1.       , 0.        , 0.         ]])
+        >>> r.as_mrp().shape
+        (1, 3)
+
+        Represent multiple rotations:
+
+        >>> r = R.from_rotvec([[np.pi/2, 0, 0], [0, 0, np.pi/2]])
+        >>> r.as_mrp()
+        array([[0.41421356, 0.        , 0.        ],
+               [0.        , 0.        , 0.41421356]])
+        >>> r.as_mrp().shape
+        (2, 3)
+
+        Notes
+        -----
+
+        .. versionadded:: 1.6.0
+        """
+        cdef Py_ssize_t num_rotations = len(self._quat)
+        cdef double[:, :] mrps = _empty2(num_rotations, 3)
+        cdef int sign
+        cdef double denominator
+
+        for ind in range(num_rotations):
+
+            # Ensure we are calculating the set of MRPs that correspond
+            # to a rotation of <= 180
+            sign = -1 if self._quat[ind, 3] < 0 else 1
+
+            denominator = 1 + sign * self._quat[ind, 3]
```

Comment:
I suggest adding a test which will show that a returned MRPs will be correct and have magnitude <= 180 degrees, when the rotation was created with magnitude >= 180 degrees (through Euler or rotation vector). The scalar quaternion component must be negative in this case.

Reasoning - to test that this mathematical transformation does in fact work, because it's not obvious.

Reply:
I think that's a good idea, test added